### ISSUE TITLE
`multipleFileExtensions` in `unicorn/filename-case` is not working correctly with eslint flat config

### ISSUE THREAD

**Issue by Geloosa** [sindresorhus/eslint-plugin-unicorn#2306]
eslint `8.57.0`
eslint-plugin-unicorn `51.0.1`
rule `unicorn/filename-case`

When using a flat config, `multipleFileExtensions` option is not [`true` by default](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/filename-case.md#multiplefileextensions) and can't be set to `true`.

This config

```javascript
{
  files: [
      "**/*.stories.js",
      "**/*.stories.ts",
      "**/*.view.test.js",
      "**/*.view.test.ts",
    ],
  rules: {
      "unicorn/filename-case": [
        "error",
        { case: "pascalCase" },
      ],
    },
}
``` 
outputs this error (though it should't)
```
/MobileMenu.stories.js
  1:1  error  Filename is not in pascal case. Rename it to `MobileMenu.Stories.js`  unicorn/filename-case

```

When trying to set `multipleFileExtensions` to `true` explicitly:

```javascript
{
  files: [
      "**/*.stories.js",
      "**/*.stories.ts",
      "**/*.view.test.js",
      "**/*.view.test.ts",
    ],
  rules: {
      "unicorn/filename-case": [
        "error",
        { case: "pascalCase", multipleFileExtensions: true },
      ],
    },
}
``` 
eslint doesn't run throwing a config validation error:
```
Oops! Something went wrong! :(

Error: Key "rules": Key "unicorn/filename-case":        Value {"case":"pascalCase","multipleFileExtensions":true} should NOT have additional properties.
        Value {"case":"pascalCase","multipleFileExtensions":true} should NOT have additional properties.
        Value {"case":"pascalCase","multipleFileExtensions":true} should match exactly one schema in oneOf.
```

**Comment by EoleO** [comment]
I have got the same problem but on legacy config.

eslint `8.57.1`
eslint-plugin-unicorn `56.0.0`
rule `unicorn/filename-case`

This config

```js
module.exports = {
  // ...
  overrides: [
    // ...
    {
      files: ["**/*.tsx"],
      rules: {
        "unicorn/filename-case": [
          "error",
          {
            case: "pascalCase",
          },
        ],
      },
    },
    // ...
  ],
  // ...
}
```

outputs this error (though it should't)
```
/AboutModal.test.tsx
  1:1  error  Filename is not in pascal case. Rename it to `AboutModal.Test.tsx`  unicorn/filename-case
```

The default value `multipleFileExtensions: true` doesn't seem to work.

**Comment by sindresorhus** [comment]
Closing as I don't think this is an issue anymore.
